### PR TITLE
Refactor `gui_templates.h`

### PR DESCRIPTION
**Pull request by Rossmaxx**
I started off trying to remove the `QApplication::desktop` call per #6614 but I ended up doing a refactor.

I removed the duplicate `pointSizeF` function and merged it into the `pointSize` function. 

There's a chance that this PR might improve HiDPI support as I did some changes in that area but haven't tested.

Todo: delete the file and find a new home for the pointSize function (should we really keep a file for a single function?)

inviting @michaelgregorius for review

### CODE REVIEW

File: include/gui_templates.h
Comment (michaelgregorius) [inline include/gui_templates.h:47]: I would remove this template as it places an unnecessary burden on the callers.

You can just declare this function as
```
inline QFont pointSize(QFont _f, float size)
```
and remove the cast in the call to `setPointSizeF`.

It's then up to the caller to put something in the function that's convertible to a `float`. And that's already the case for all clients that make use of the function. For example, instead of doing the call
```
setFont(pointSize<int>(m_interpBox->font(), 8));
```
the client would call:
```
setFont(pointSize(m_interpBox->font(), 8));
```
Or if we want to "help" the compiler a bit:
```
setFont(pointSize(m_interpBox->font(), 8.));
```

Comment (Rossmaxx) [reply include/gui_templates.h]: done.

**Comment by michaelgregorius** [comment]
Looks good to me! :+1:

Concerning the new home for the functionality: is there some kind of GUI helpers file where it could move?

File: include/gui_templates.h
Comment (Snowiiii) [inline include/gui_templates.h:51]: i think we should use `std::max(1.0, devicePixelRatio)` for better readability

File: include/gui_templates.h
Comment (Snowiiii) [inline include/gui_templates.h:50]: When using `QGuiApplication::primaryScreen()` i think there could be some issues when Switching the App to an other Monitor with an diffrent DPI, We should grab the current Monitor for our Window

File: include/gui_templates.h
Comment (Rossmaxx) [reply include/gui_templates.h:51]: Why didn't i think of this?

File: include/gui_templates.h
Comment (Rossmaxx) [reply include/gui_templates.h:50]: I get what you mean but can you suggest a way to handle this one properly?

**Comment by Rossmaxx** [comment]
> there some kind of GUI helpers file where it could move?

Looking at the sources, i didn't really find a more suitable file.

File: include/gui_templates.h
Comment (michaelgregorius) [reply include/gui_templates.h:50]: I think this is something that should be handled by the GUI framework, i.e. by Qt. It just one example of why using these functions in application code is problematic.

Using code like this also won't do the "right thing" when LMMS is in the middle of being dragged from one screen to another with different DPI values. Qt likely knows that one part of the application rectangle must be rendered with different settings than the other and can do so. The application code does not even have the knowledge of the current state ("I am shown on two different monitors.") and cannot do the right thing.

**Comment by michaelgregorius** [comment]
> > there some kind of GUI helpers file where it could move?
> 
> Looking at the sources, i didn't really find a more suitable file.

If it only contains this single function then another option might be to rename the file to `HiDpiHelper.h` or similar. The goal might then even be to adjust all places such that the file becomes superfluous and can be removed. The file would then reflect the ongoing transition to make LMMS work on HiDPI screens.

**Comment by Rossmaxx** [comment]
I'll hold off on the rename for now, to reduce the diff.

File: include/gui_templates.h
Comment (Rossmaxx) [reply include/gui_templates.h]: done with https://github.com/LMMS/lmms/pull/7159/commits/296583e8d3cd676b171b86ea7135663d11acf0a7

**Review by michaelgregorius — APPROVED** [review]

**Comment by Rossmaxx** [comment]
I am thinking of moving this to the gui namespace but i am doubtful about the changes I need to make in plugin code. + I have some other minor changes in mind which I'll test soon.

**Comment by michaelgregorius** [comment]
> I am thinking of moving this to the gui namespace but i am doubtful about the changes I need to make in plugin code. + I have some other minor changes in mind which I'll test soon.

Such a move would definitively make sense. The code should only be used from GUI code and not from core code anyway.

**Review by messmerd — APPROVED** [review]

**Comment by Rossmaxx** [comment]
This pr got 2 approvals but it ain't ready for merge yet. I got a few more minor cleanups to do. So can you wait for an extra day.

**Comment by Veratil** [comment]
> This pr got 2 approvals but it ain't ready for merge yet. I got a few more minor cleanups to do. So can you wait for an extra day.

Mark it as a draft and it will prevent merge 🙂

**Comment by Rossmaxx** [comment]
I'm done with my changes, let the ci run before i make this pr ready for review.

**Pull request merged**
